### PR TITLE
updated objects to class

### PR DESCRIPTION
**Pull request by clarakosi**
* Updated objects in `lib/` to follow javascript class format 
* Changed `no-use-before-define` eslint rule to warning from error in `.eslintrc.yml`

### CODE REVIEW

**Comment by Pchelolo** [comment]
hm... why aint travis picking it up..

**Comment by clarakosi** [comment]
> hm... why aint travis picking it up..

Not sure

File: .eslintrc.yml [hunk 1]
```diff
@@ -2,3 +2,4 @@ extends: 'eslint-config-node-services'
 
 rules: 
   no-bitwise: 0
+  no-use-before-define: 1
```
Comment (Pchelolo) [inline .eslintrc.yml]: We actually set it up in the `eslint-config-node-services` https://github.com/wikimedia/eslint-config-node-services/blob/master/.eslintrc.js#L60

This means that we allow using classes before they're declared. So if you move the createUuidFromBuffer function to the top of the file, you will be able to delete this line.

**Review by Pchelolo — CHANGES_REQUESTED** [review]

**Comment by Pchelolo** [comment]
One minor nit inlined, otherwise LGTM. Let's also bump a version and release the package

**Review by Pchelolo — DISMISSED** [review]

**Comment by Pchelolo** [comment]
All right! LGTM. Let's bump a minor version, it's a big set of changes and publish.

File: package-lock.json [hunk 2]
```diff
@@ -0,0 +1,2850 @@
+{
```
Comment (Pchelolo) [inline package-lock.json]: Atatata... Remember what we've talked about regading the package-lock?

Let's remove the file and add it to .gitignore for future convinience.

Comment (Pchelolo) [reply package-lock.json]: https://git-scm.com/docs/gitignore - that's what the gitignore is. It's usually used for things you want to have in the project directory, but don't want to commit. Normally that would be database files, dependencies, configs etc.

**Comment by Pchelolo** [comment]
A righty. I gave you the rights to publish this to npm. Could you please do that?

**Pull request merged**